lib/fluent/plugin/out_elasticsearch_data_stream.rb

```diff
@@ -0,0 +1,169 @@
+require_relative 'out_elasticsearch'
+
+module Fluent::Plugin
+  class ElasticsearchOutputDataStream < ElasticsearchOutput
+
+    Fluent::Plugin.register_output('elasticsearch_data_stream', self)
+
+    helpers :event_emitter
+
+    config_param :data_stream_name, :string,  :default => ''
+
+    INVALID_START_CHRACTERS = ["-", "_", "+", "."]
+    INVALID_CHARACTERS = ["\\", "/", "*", "?", "\"", "<", ">", "|", " ", ",", "#", ":"]
+
+    def configure(conf)
+      super
+
+      if @data_stream_name.empty?
+        raise Fluent::ConfigError, "'data_stream_name' must not be empty."
```

Comment:
Should we use `Fluent::ConfigError` instead of mandatory parameter syntax like this?

```ruby
    config_param :data_stream_name, :string
```

I think that we shouldn't use runtime checking if strongly needed.
In this case, we should use mandatory parameter syntax.

And when we use mandatory parameter syntax, we can generate required attributes automatically with `fluent-plugin-config-format`.

Reply:
Thanks, 
https://www.clear-code.com/blog/2017/3/9.html